src/store/modules/app-gateway-svc/mutations.ts

```diff
@@ -0,0 +1,15 @@
+import { MutationTree } from 'vuex';
+import { ModuleState } from './types';
+import * as mutation from './types-mutations';
+import { AppGatewayServiceClient, grpc } from '../../root-types';
+
+// eslint-disable-next-line import/prefer-default-export
+export const mutations: MutationTree<ModuleState> = {
+  [mutation.SET_HTTP_HEADER](state, payload: grpc.Metadata) {
+    Object.assign(state, { httpHeader: payload });
+    // state.httpHeader = payload;
```

Comment:
We should probably rename the `state` param to maybe `newState` ?

Reply:
It's either a convention to use `state` or it actually refers to the `export const state` object we defined in `states.ts`. I know in `actions.ts`, `state` definitely refers to the defined state object and can't change the name.